benchmarks/persistence/persistence.py

```diff
@@ -0,0 +1,210 @@
+"""Classes for persisting benchmark results."""
+
+import json
+import os
+from pathlib import Path
+from typing import Protocol
+
+import boto3
+import boto3.session
+from attr import define, field
+from attrs.validators import instance_of
+from git import Repo
+
+from benchmarks import Benchmark, Result
+
+MODULE_IS_RUNNING_IN_THE_PIPELINE = "GITHUB_RUN_ID" in os.environ
+
+
+class PersistenceObjectInterface(Protocol):
+    """Interface for constructing the path of a result object."""
+
+    def get_full_file_path(self) -> str:
+        """Construct the path of a result object.
+
+        Returns:
+            The path of the result object.
+        """
+
+
+class PersistenceHandlingInterface(Protocol):
+    """Interface for persisting experiment results."""
+
+    def write_object(self, path: PersistenceObjectInterface, object: dict) -> None:
```

Comment:
I am confused here - the argument `path` is not a `Path`, but a `PersistenceObjectInterface`, that feels weird.

Reply:
@AVHopp I have changed the logic after a discussion with @AdrianSosic , please let me know what you think :)